### DISCUSSION TITLE
Bottom axis valueformatting crash

### DISCUSSION THREAD

**Discussion by siroaleksi** [patrykandpatrick/vico#937] [Questions]
### Question

Hi, I have implemented a two series line chart using vico 2.0.0-beta.1 I need to display either measurement time or measurement day in the x-axis depending on chart style. I have converted millis into String timestamp e.g. "10:00" or "Oct 1" before doing anything with the chart. 

I have created labelListKey like this `val labelListKey = remember { ExtraStore.Key<List<String>>() }` I have used remeber because without it I get null pointer exception. Then I have created key value pairs of data like `val data = xLabels.zip(data.map { it.toFloat() }).toMap()` Then I update extras and series as 

```
modelProducer.runTransaction {  
   lineSeries {  
      series(y = data1)  
      series(y = data2)  
   }  
   extras { it[labelListKey] = data.keys.toList() }  
}
```  
Series share same x-labels. Finally in the bottomAxis I have 

```  
valueFormatter = { context, x, _ ->
   Log.i("X", "x: $x, key $labelListKey")
   context.model.extraStore[labelListKey][x.toInt()]
}
```  

I have a segmented button row where I select different chart style Day, Week and Month, then I gather data from Room db before running ` modelProducer.runTransaction `.
 
Everything works well if I change style with moderate pace. If I rapidly change style, the app eventually crash. 
When logging x value in the valueFormatter I get something like this. 

```
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 7.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 4.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 7.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 7.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 4.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 7.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
```
before x values start increasing normally 

```
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 1.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 2.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 3.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 4.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 5.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 6.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 7.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
```

At some point I get this 
```
x: 0.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089
x: 30.0, key com.patrykandpatrick.vico.core.common.data.ExtraStore$Key@f32f089

java.lang.IndexOutOfBoundsException: Index 30 out of bounds for length 20
```

Am I somehow handling data incorrectly in the
```
modelProducer.runTransaction {  
   lineSeries {  
      series(y = data1)  
      series(y = data2)  
   }  
   extras { it[labelListKey] = data.keys.toList() }  
}
```  
which lead to this random x value before normal incremental? I have checked that the sizes of xlabels, data1 and data 2 matches before crash.

### Vico version(s)

2.0.0-beta.1

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment]
Hello, @siroaleksi. The `runTransaction` usage looks correct, but there may be an issue elsewhere. Please share a complete, runnable example that demonstrates the crash so we can help.
